staging/src/k8s.io/apiserver/pkg/util/flowcontrol/bootstrap/default-objects.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"math"
+
+	rmv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// The objects that define an apiserver's initial behavior
+var (
+	// InitialPriorityLevelConfigurations lists exempt first, default second
+	InitialPriorityLevelConfigurations = []*rmv1a1.PriorityLevelConfiguration{
+		DefaultPriorityLevelConfigurationExempt,
+		DefaultPriorityLevelConfigurationDefault,
+	}
+	InitialFlowSchemas = []*rmv1a1.FlowSchema{
+		DefaultFlowSchemaExempt,
+		DefaultFlowSchemaDefault,
+	}
+)
+
+// Initial PriorityLevelConfiguration objects
+var (
+	DefaultPriorityLevelConfigurationExempt = pl(
+		rmv1a1.PriorityLevelConfigurationNameExempt,
+		nil)
+
+	DefaultPriorityLevelConfigurationDefault = plq(
+		rmv1a1.PriorityLevelConfigurationNameCatchAll, 100,
+		&rmv1a1.QueuingConfiguration{
+			Queues:           128,
+			HandSize:         6,
+			QueueLengthLimit: 100,
+		})
+)
+
+// Initial FlowSchema objects
+var (
+	DefaultFlowSchemaExempt = NewFSAllGroups(
+		rmv1a1.FlowSchemaNameExempt,
+		rmv1a1.PriorityLevelConfigurationNameExempt,
+		1, "",
+		user.SystemPrivilegedGroup,
+	)
+	DefaultFlowSchemaDefault = NewFSAllGroups(
+		rmv1a1.FlowSchemaNameCatchAll,
+		rmv1a1.PriorityLevelConfigurationNameCatchAll,
+		math.MaxInt32, rmv1a1.FlowDistinguisherMethodByUserType,
+		user.AllUnauthenticated, user.AllAuthenticated,
+	)
+)
+
+// PredefinedPriorityLevelConfigurations returns the instances of
+// the ones defined in the KEP
+// with the exempt one appearing first
+func PredefinedPriorityLevelConfigurations() []*rmv1a1.PriorityLevelConfiguration {
+	return []*rmv1a1.PriorityLevelConfiguration{
+		DefaultPriorityLevelConfigurationExempt,
+		spl(plq("system", 30, &rmv1a1.QueuingConfiguration{
+			Queues:           1,
+			QueueLengthLimit: 1000})),
+		spl(plq("workload-high", 30, &rmv1a1.QueuingConfiguration{
+			Queues:           128,
+			HandSize:         6,
+			QueueLengthLimit: 100})),
+		spl(plq("workload-low", 30, &rmv1a1.QueuingConfiguration{
+			Queues:           128,
+			HandSize:         6,
+			QueueLengthLimit: 100})),
+		DefaultPriorityLevelConfigurationDefault,
+	}
+}
+
+var (
+	verbAll       = []string{rmv1a1.VerbAll}
+	apiGroupAll   = []string{rmv1a1.APIGroupAll}
+	apiGroupCore  = ""
+	nsAll         = []string{"c s", "*"}
```

Comment:
does `c s` stand for cluster-scoped? if so, we are supposed to do it by the new boolean field in the resource-rule spec

Reply:
This is just a hack for the construction convenience function in this file.